_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,371 @@
+---
+id: d0f69e
+name: All table header cells have assigned data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has assigned data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgements:
+  authors:
+    - Jey Nandakumar
+    - Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any HTML element that
+
+- has the [semantic role][] of [rowheader][] or [columnheader][] and;
+- is [visible][] and;
+- is [included in the accessibility tree][] and;
+- whose closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]:
```

Comment:
```suggestion
- has at least one ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]; and
- whose closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]:
```
Otherwise, this cannot be evaluated on Inapplicable example 6.